### PR TITLE
PoC code to get disk metrics from node

### PR DESCRIPTION
**Pull request by kieron-dev**
In Eirini team we are maintaining [a component for metrics](https://github.com/cloudfoundry-incubator/eirini/tree/master/cmd/metrics-collector) which isn't used in cf-for-k8s. We would like to deprecate that component but the [one used in cf-for-k8s](https://github.com/cloudfoundry/metric-proxy) doesn't support disk usage yet.

This PR is a PoC that makes disk usage metrics work in metric-proxy.

TODOs:
- Fix tests
- Add tests for new functionality
- Possibly add caching for the fetched metrics since they contain information for more than the requested pod and thus could be reused.

More to consider:

The current implementation doesn't add the image size in the calculated total. That means that an application like [dora](https://github.com/cloudfoundry/cf-acceptance-tests/tree/develop/assets/dora) when first pushed it appears as if it consumes just 12K of disk, although there are a number of gems obviously pulled in (but are part of the docker image).

The image sizes are available under the "/api/v1/nodes" endpoint.

You can check with:

```
$ # Start a proxy to the kubernetes api
$ kubectl proxy --api-prefix=/
$ curl localhost:8001/api/v1/nodes | jq '.items[0].status.images' --color-output | less -R
```

https://www.pivotaltracker.com/story/show/175935123

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -15,17 +16,17 @@ import (
...
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/metrics/pkg/apis/metrics/v1beta1"
 	"k8s.io/metrics/pkg/client/clientset/versioned"
 )
 
 var (
-	version = "dev-build"
+	version          = "dev-build"
```
Comment (jamespollard8) [inline main.go:26]: whoops?

**Review by jamespollard8 — COMMENTED** [review]
Looks good so far. Having this functionality would be great!

We look forward to seeing future iterations of this PR.

File: main.go (hunk 1 above)
Comment (kieron-dev) [reply main.go:27]: We're using gofumpt to format.

**Comment by gcapizzi** [comment]
Hey @jamespollard8! We think we have made all the necessary changes to this PR, can you review this again? Thanks!

File: pkg/metrics/proxy.go [hunk 2]
```diff
@@ -126,7 +151,7 @@ func (m *Proxy) createLoggregatorEnvelope(
 }
 
 func (m *Proxy) createGaugeMap(k v1.ResourceName, v resource.Quantity) map[string]*loggregator_v2.GaugeValue {
-	var gauges = map[string]*loggregator_v2.GaugeValue{}
+	gauges := map[string]*loggregator_v2.GaugeValue{}
```
Comment (jspawar) [inline pkg/metrics/proxy.go:154]: Small suggestion, doubt it matters though: probably could replace this with something like `gauges := make(map[string]...)`?

File: tools/tools.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// +build tools
```
Comment (jspawar) [inline tools/tools.go:1]: What is this file for?

**Review by jspawar — COMMENTED** [review]
Thanks, looks good, going to test changes before merging. Happy to see more convergence in the CF on K8s space

**Comment by jspawar** [comment]
Ran CATS and smoke tests against a cf-for-k8s installation with these changes via GitHub actions on a dummy PR: https://github.com/cloudfoundry/cf-for-k8s/pull/618

They all passed so this looks ready to merge. Some answers on my most recent comments would be great, but will go ahead and merge tomorrow regardless

File: pkg/metrics/proxy.go (hunk 2 above)
Comment (kieron-dev) [reply pkg/metrics/proxy.go:154]: Eek - I think that was `gofumpt` (our formatter) automatically changing that - so it wasn't something we deliberately changed from the original code. I suppose make is better if you know the expected size to reduce a couple of allocations as the map grows.

File: tools/tools.go (hunk 3 above)
Comment (kieron-dev) [reply tools/tools.go:1]: This creates a dependency on counterfeiter for the project. It has the effect that when you `go generate` to produce the code for the test fakes, each developer will use an identical version of counterfeiter as given in the module configuration. So it's an improvement on having counterfeiter as an external dependency where devs can have different versions producing difference code.

**Pull request merged**